### PR TITLE
define new entry points for chemical composition v2

### PR DESCRIPTION
**Pull request by StSchulze**
Defined new entry points for chemical composition, to explicitly define peptide sequences, chemical formulas, modifications and glycans:
> sequence (str): peptide sequence that should be used for the instance, e.g. "PEPTIDE"
> formula (str): chemical formula(s) that will be added/subtracted from the instance, e.g. "-H2O"
> modifications (str): modifications of the peptide sequence, given as "Unimod:pos", e.g. "Oxidation:2"
> glycan (str): glycan composition that will be added to the instance, e.g. "HexNAc(2)Hex(5)"
> deprecated_format (str): old input format, using either "PEPTIDE+Formula-Formula" or "PEPTIDE#Unimod:pos"

### CODE REVIEW

**Comment by StSchulze** [comment]
We should still add more tests, but the current ones cover most of the previous functionality and most of the new functions.

Let me know if this style for the entry points is fine.

File: chemical_composition/chemical_composition.py [hunk 1]
```diff
@@ -161,10 +159,95 @@ def clear(self):
...
+            self.parse_chemical_formula(formula)
+
+        if deprecated_format is not None:
+            if (
+                sequence is not None
+                or formula is not None
+                or modifications is not None
+                or glycan is not None
+            ):
+                sys.exit(
```
Comment (MKoesters) [inline chemical_composition/chemical_composition.py:215]: I'd always prefer to raise an Exception instead of sys.exit

**Review by MKoesters — APPROVED** [review]

File: chemical_composition/chemical_composition.py (hunk 1 above)
Comment (StSchulze) [reply chemical_composition/chemical_composition.py:215]: Fair, I'll change it. That goes for all sys.exits in here, right?

Comment (MKoesters) [reply chemical_composition/chemical_composition.py]: Yes, I'd prefer to use Exceptions, especially in a library, in a CLI program there are some good reasons.
Thing is, you can't catch a `sys.exit()` with `execept Exception:`, only with an `except:`

**Pull request merged**